File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
const ColorMap: Record<string, number> = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
};


export function decodedValue(colors: string[]): number {
  // 1. Take the first two elements of the array
  // 2. Map the names to their numeric values
  // 3. Join them into a string and convert back to a number
  
  const [first, second] = colors;
  
  const val1 = ColorMap[first.toLowerCase()];
  const val2 = ColorMap[second.toLowerCase()];

  // Combine the digits (e.g., 1 and 5 becomes 15)
  return val1 * 10 + val2;
}

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
def value(colors):
    band_values = [
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    ]
    
    first_color = colors[0]
    second_color = colors[1]
    
    first_digit = band_values.index(first_color)
    second_digit = band_values.index(second_color)
    
    return first_digit * 10 + second_digit